### DISCUSSION TITLE
CoreTextServicesManager not working

### DISCUSSION THREAD

**Discussion by FrozenAssassine** [microsoft/WindowsAppSDK#3226] [Q&A] [answered]
I want to create a custom textbox for the WindowsAppSdk, and came across the CoreTextServicesManager. I tried it, but it did not work, what am I doing wrong?

```
CoreTextServicesManager manager = CoreTextServicesManager.GetForCurrentView();
EditContext = manager.CreateEditContext();
EditContext.InputPaneDisplayPolicy = CoreTextInputPaneDisplayPolicy.Manual;
EditContext.InputScope = CoreTextInputScope.Text;
EditContext.TextRequested += delegate { };
EditContext.SelectionRequested += delegate { };
EditContext.TextUpdating += EditContext_TextUpdating;
EditContext.FocusRemoved += EditContext_FocusRemoved;

EditContext.NotifyFocusEnter(); 

private void EditContext_TextUpdating(CoreTextEditContext sender, CoreTextTextUpdatingEventArgs args)
{
   Debug.WriteLine(args.Text);
}
private void EditContext_FocusRemoved(CoreTextEditContext sender, object args)
{
    Debug.WriteLine("Lost focus");
}
```

**Comment by DarranRowe** [comment, marked as answer]
This is expected. Anything that uses GetForCurrentView is basically getting an instance of this class attached to a CoreWindow. The issue is that CoreWindow isn't available at all in a desktop environment, which is what WinUI/Windows App SDK runs in.

What this means is these runtime classes need some method to get an instance of the class attached to a HWND instead. [There is documentation](https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/desktop-to-uwp-supported-api#classes-with-an-xxxforcurrentview-method) stating which of these runtime classes can currently be used in desktop applications and what you need to do to get an instance, but as you can imagine, it isn't that easy.

According to this documentation, CoreTextServiceManager isn't supported in any release version of Windows and is only currently supported in the preview version of Windows. What's more, the [documentation that lists the API mapping](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/api-mapping-table) states that this is only available on Windows 11. So my thought is that currently there is no support for what you want to do and it will only become available in the next update for Windows 11.
